### PR TITLE
Update /etc/st2.conf to match the new options available in the st2 package

### PR DESCRIPTION
**Pull request by jvrplmlmn**
Fixes stackstorm/chef-stackstorm#14

@armab @shortdudey123 @sysbot ?

### CODE REVIEW

File: templates/default/st2.conf.erb [hunk 1]
```diff
@@ -65,12 +84,6 @@ remote_dir = /tmp
 [action_sensor]
 triggers_base_url = <%= @api_url %>/v1/triggertypes/
 
-[resultstracker]
-logging = <%= @conf_root %>/st2actions/console.conf
-
-[notifier]
-logging = <%= @conf_root %>/st2actions/console.conf
-
 [database]
```
Comment (jvrplmlmn) [inline templates/default/st2.conf.erb:87]: @armab Dunno if this is still relevant with `2.0.0` or we can get rid of this `database` category.

**Comment by estee-tew** [comment]
https://stackstorm.slack.com/archives/community/p1473186082000207

File: attributes/config.rb [hunk 2]
```diff
@@ -3,8 +3,12 @@
 # Config defaults options
 default['stackstorm']['config'] = {
   api_url: 'http://localhost:9101',
-  conf_root: '/etc',
+  api_host: '0.0.0.0',
```
Comment (lakshmi-kannan) [inline attributes/config.rb:6]: Shouldn't we bind to 127.0.0.1? Binding to all interfaces seems a little dangerous.

File: templates/default/st2.conf.erb (hunk 1 above)
Comment (lakshmi-kannan) [reply templates/default/st2.conf.erb:87]: You can remove this. https://github.com/StackStorm/st2/blob/master/conf/st2.prod.conf. 

If not provided, default host is 127.0.0.1 and port to connect would be 27017.

File: attributes/config.rb (hunk 2 above)
Comment (arm4b) [reply attributes/config.rb:6]: @lakshmi-kannan That thing comes from the https://github.com/StackStorm/st2/blob/master/conf/st2.package.conf#L5

If that's new recommended value, I think we should edit `st2.package.conf` and `st2.prod.conf` first.

And Chef recipe will follow default values from `st2.conf` as source of truth.

Comment (lakshmi-kannan) [reply attributes/config.rb:6]: Good catch @armab. Maybe open a PR in st2? ;)

Comment (arm4b) [reply attributes/config.rb:6]: Was your catch :stuck_out_tongue_winking_eye:

**Comment by arm4b** [comment]
Looks like Travis complains because of hardcoded test cases in: https://github.com/StackStorm/chef-stackstorm/blob/master/spec/recipes/config_spec.rb related to old `st2.conf` settings.

File: templates/default/st2.conf.erb (hunk 1 above)
Comment (shortdudey123) [reply templates/default/st2.conf.erb:87]: This should be left, otherwise it is not possible to use a mongo cluster outside of localhost

**Comment by shortdudey123** [comment]
ChefSpec tests need to be updated and the file should follow the same order as the packaged one.  Looks like some stuff got out of order in this PR

**Comment by arm4b** [comment]
Seems this wasn't updated for a while.
Will open another PR to address all `st2.conf` changes.

**Pull request closed without merging**